### DISCUSSION TITLE
Select of Material-ui doesn't set value

### DISCUSSION THREAD

**Discussion by Gumichocopengin8** [react-hook-form/react-hook-form#1690] [General] [answered]
Hello, 
Thank you for making a great form library! It is super helpful.

I got stuck with select of material-ui

```
<Controller
              name="category"
              as={
                <FormControl variant="outlined" error={!!errors.majorCategory}>
                  <Select>
                    {items.map((item) => (
                      <MenuItem key={item.key} value={item.value}>
                        {item.text}
                      </MenuItem>
                    ))}
                  </Select>
                </FormControl>
              }
              control={control}
              rules={{ required: true }}
              onChange={([selected]) => selected}
              valueName="selected"
            />
```
When I console out `getValues('category')`, it doesn't not return selected value. 
How can I fix it?? I want to keep using `Controller` if possible.

Thank you!

**Comment by bluebill1049** [comment, marked as answer]
hey @Gumichocopengin8 can you create a simple codesandbox for this problem?

**Reply by Gumichocopengin8** [reply]
Thanks for quick reply.
here is the code
https://codesandbox.io/s/stupefied-bouman-vstrs?file=/src/App.tsx

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/happy-pare-95sqc?file=/src/App.tsx

`FormControl` doesn't have `onChange` props.

**Reply by Gumichocopengin8** [reply]
Yeah, but i want to use `FormControl` because it has error props.
In this case, I need to use `setValue`?

**Reply by bluebill1049** [reply]
did you see my CSB? you just need to wrap it outside.

**Reply by Gumichocopengin8** [reply]
Oh yeah... I completely forgot about it. 
Thank you very much!
